pkg/ipmi/oem.go

```diff
@@ -0,0 +1,333 @@
+// Copyright 2017-2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package ipmi
+
+import (
+	"bytes"
+	"encoding/binary"
+	"strconv"
+	"strings"
+	"unsafe"
+
+	"github.com/u-root/u-root/pkg/smbios"
+)
+
+const (
+	_IPMI_FB_OEM_NET_FUNCTION = 0x36
+
+	//OEM Commands
+	_FB_OEM_SET_PROC_INFO = 0x10
+	_FB_OEM_SET_DIMM_INFO = 0x12
+)
+
+type oemData interface {
+	data()
+}
+
+type oemProcProductNameDataParam1 struct {
```

Comment:
How about doing something similar to the Event type in: https://github.com/u-root/u-root/blob/master/pkg/ipmi/ipmi.go#L136 and having the shared fields on a base type.

Reply:
Done.